### DISCUSSION TITLE
Prevent auto connect to metamask

### DISCUSSION THREAD

**Discussion by ale-ben** [WalletConnect/walletconnect-monorepo#4014] [Developer Support] [answered]
I am following the NextJS Wagmi getting started guide (https://docs.walletconnect.com/web3modal/nextjs/about).
I have a `<w3m-button />` that opens the modal to select wallet and connect.

Each time I delete the cache and reload the page, the web app tries to connect automatically to MetaMask.
I want the connection to MetaMask to happen after the user opens the  `<w3m-button />` modal, not as soon as the web page loads.
I know that disabling auto-connect would mean that after each reload the user would have to open the modal but I prefer it rather than having the metamask popup open as soon as the home page first loads.

The only thing I found is this issue: https://github.com/WalletConnect/web3modal/pull/1515. 

Am I missing something?

**Comment by glitch-txs** [comment, marked as answer]
Hi, we've fixed this issue, and it's going to be included in the next release

**Reply by codemonkei** [reply]
@glitch-txs is this fix already released as the latest version (https://github.com/WalletConnect/web3modal/releases/tag/3.5.3) still keeps the popping up metamask:

the metamask plugin automatically opens when you are reopening the browser and heading to a prev. connected page like https://web3modal.com/. Is this expected behaviour as we think in web3 nothing should auto-pop without a user interaction. 

This is our client side config: 

const { chains, publicClient } = configureChains(
  [mainnet],
  [walletConnectProvider({ projectId }), publicProvider()]
);

const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: [
    new WalletConnectConnector({
      chains,
      options: { projectId, showQrModal: false, metadata },
    }),
    new EIP6963Connector({ chains }),
    new InjectedConnector({ chains, options: { shimDisconnect: true } }),
    new CoinbaseWalletConnector({
      chains,
      options: { appName: metadata.name },
    }),
  ],
  publicClient,
});

createWeb3Modal({ wagmiConfig, projectId, chains });

**Reply by glitch-txs** [reply]
> in web3 nothing should auto-pop without a user interaction.

This is correct. 💯 

Can you try clearing cache? It should be fixed, let me know if after clearing cache you're still able to reproduce the issue. Thanks!

**Reply by glitch-txs** [reply]
nvm, I found the issue, we'll investigate it.

**Reply by glitch-txs** [reply]
Hi @doktordos-eth it should be fixed now I'm the 3.5.4 version

**Reply by codemonkei** [reply]
@glitch-txs can confirm that it works now, thanks for the fast fix 👍🏽

**Comment by aminpsycho** [comment]
ok

**Comment by ProfBeier** [comment]
thanks
